### PR TITLE
DP-888 feat: added support for large video upload

### PR DESCRIPTION
**Pull request by krgauraw**

### CODE REVIEW

File: platform-jobs/samza/auto-creator/src/main/java/org/ekstep/jobs/samza/util/ContentUtil.java [hunk 1]
```diff
@@ -214,17 +224,46 @@ private String create(String channelId, String identifier, String repository, Ma
 	}*/
 
 	private void upload(String channelId, String identifier, Map<String, Object> metadata) throws Exception {
+		Response resp = null;
+		Long downloadStartTime = System.currentTimeMillis();
+		LOGGER.info("ContentUtil :: upload :: Starting File Download at: " + downloadStartTime);
```
Comment (maheshkumargangula) [inline platform-jobs/samza/auto-creator/src/main/java/org/ekstep/jobs/samza/util/ContentUtil.java:229]:
```suggestion
```

File: platform-jobs/samza/auto-creator/src/main/java/org/ekstep/jobs/samza/util/ContentUtil.java [hunk 2]
```diff
@@ -214,17 +224,46 @@ private String create(String channelId, String identifier, String repository, Ma
...
 		File file = getFile(identifier, (String) metadata.get(AutoCreatorParams.artifactUrl.name()));
+		Long downloadEndTime = System.currentTimeMillis();
+		LOGGER.info("ContentUtil :: upload :: Finished File Download at: " + downloadEndTime);
```
Comment (maheshkumargangula) [inline platform-jobs/samza/auto-creator/src/main/java/org/ekstep/jobs/samza/util/ContentUtil.java:232]:
```suggestion
```

File: platform-jobs/samza/auto-creator/src/main/java/org/ekstep/jobs/samza/util/ContentUtil.java [hunk 3]
```diff
@@ -214,17 +224,46 @@ private String create(String channelId, String identifier, String repository, Ma
...
 			url = url + "&fileFormat=composed-h5p-zip";
 		Map<String, String> header = new HashMap<String, String>() {{
 			put("X-Channel-Id", channelId);
 		}};
-		Response resp = UnirestUtil.post(url, "file", file, header);
+
+		if (size > CONTENT_UPLOAD_ARTIFACT_MAX_SIZE) {
+			if (BULK_UPLOAD_MIMETYPES.contains(mimeType)) {
+				Long uploadStartTime = System.currentTimeMillis();
+				LOGGER.info("ContentUtil :: upload :: Starting File Upload at: " + uploadStartTime);
```
Comment (maheshkumargangula) [inline platform-jobs/samza/auto-creator/src/main/java/org/ekstep/jobs/samza/util/ContentUtil.java:250]:
```suggestion
```

File: platform-jobs/samza/auto-creator/src/main/java/org/ekstep/jobs/samza/util/ContentUtil.java [hunk 4]
```diff
@@ -214,17 +224,46 @@ private String create(String channelId, String identifier, String repository, Ma
...
+				String[] urls = uploadArtifact(file, identifier);
+				Long uploadEndTime = System.currentTimeMillis();
+				LOGGER.info("ContentUtil :: upload :: Finished File Upload at: " + uploadEndTime);
```
Comment (maheshkumargangula) [inline platform-jobs/samza/auto-creator/src/main/java/org/ekstep/jobs/samza/util/ContentUtil.java:253]:
```suggestion
```

File: platform-jobs/samza/auto-creator/src/main/java/org/ekstep/jobs/samza/util/ContentUtil.java (hunk 2 above)
Comment (krgauraw) [reply platform-jobs/samza/auto-creator/src/main/java/org/ekstep/jobs/samza/util/ContentUtil.java]: done

File: platform-jobs/samza/auto-creator/src/main/java/org/ekstep/jobs/samza/util/ContentUtil.java (hunk 3 above)
Comment (krgauraw) [reply platform-jobs/samza/auto-creator/src/main/java/org/ekstep/jobs/samza/util/ContentUtil.java]: done

File: platform-jobs/samza/auto-creator/src/main/java/org/ekstep/jobs/samza/util/ContentUtil.java (hunk 4 above)
Comment (krgauraw) [reply platform-jobs/samza/auto-creator/src/main/java/org/ekstep/jobs/samza/util/ContentUtil.java]: done

**Pull request merged**